### ISSUE TITLE
Hint: Delegate Allocation from Method Group Displays Even when Instantiated

### ISSUE THREAD

**Issue by Mike-EEE** [controlflow/resharper-heapview#36]
I wanted to bring this up to ensure that I am understanding this correctly and/or to confirm that this might be a bug, or at the very least misleading.

I have created a delegate using the `new` and passing the delegate in with the constructor:

![](http://content.screencast.com/users/Mike-EEEE/folders/Jing/media/353f8b7f-962b-4710-9d09-6e596be58a13/2016-06-11_1833.png)

The `Get` is still underlined, even when I use the constructor.  Is this expected?  It seems to me that by instantiating the delegate directly, that the instantiation is what causes the allocation.  That is, the hint should be under the `new` rather than the method symbol.

My other concern is that there are actually two allocations happening, but since `new` is not underlined, then it is just the allocation from the method symbol (resulting in a delegate).  I want to ensure/verify that only one allocation is happening in this scenario.

Again, I am new to all of this, so I appreciate your patience in explaining anything that I have mistaken here.  Thanks again!

**Comment by controlflow** [comment]
Hi.

1. In you example, there is only one allocation (I assume `Get` is a method of containing class, not struct) - allocation to create delegate instance. So HeapView is kinda right here;
2. I think I haven't supported explicit delegate creation expression well, since it is not oftenly used in C# code and I forgot about it. I think, HeapView should highlight `new` keyword here instead of method group expression to make things clear. I'll fix that 👍

**Comment by Mike-EEE** [comment]
Ah!  Yes, `Get` is a method of the containing class.  And awesome... glad I could help. :)

**Comment by Mike-EEE** [comment]
Here is another related scenario I just encountered.  I am sure what you have in mind will address this, but I wanted to capture it here just in case:

Given a delegate and class:
```
public delegate void ActionWithParameter( string first, bool second, DateTime third );
class EmptyClass
{
	public void Method( string first, bool second, DateTime third ) {}
}
```

In my test:

```
var empty = new EmptyClass();
var action = new Action<string, bool, DateTime>( empty.Method );
var other = new ActionWithParameter( action );
```

I get a hint under `empty.Method` but not under the `other` assignment.  Here is a screenshot of the actual code/results:
![](http://content.screencast.com/users/Mike-EEEE/folders/Jing/media/630fe180-4ab1-483d-a90a-fc2d1acbdf64/2016-06-12_1701.png)

I have looked at both references, and they do appear to be different (Make Object ID creates a new Id for each object, that is), so it would seem that the `new ActionWithParameter` is a new allocation.
